src/compiler/diagnosticMessages.json

```diff
@@ -7936,5 +7936,13 @@
     "'await using' statements cannot be used inside a class static block.": {
         "category": "Error",
         "code": 18054
+    },
+    "A member initializer in a enum declaration for a string value must be a string literal when 'isolatedModules' is enabled.": {

```

Comment:
If `evaluatesToStringLiteral` and `evaluatesToNumericLiteral` remain fairly complex, we should probably try to find more accurate wording for these messages.

Reply:
Makes sense. Something like these maybe?

- A string member initializer in a enum declaration can't include references when 'isolatedModules' is enabled.
- A string member initializer in a enum declaration can only use constant expressions when 'isolatedModules' is enabled.

Reply:
I updated the error message with the 2nd option.